components/concept/engine/src/main/java/mozilla/components/concept/engine/permission/PermissionRequest.kt

```diff
@@ -61,11 +61,13 @@ interface PermissionRequest {
  *
  * @property id an optional native engine-specific ID of this permission.
  * @property desc an optional description of what this permission type is for.
+ * @property name permission name allowing to easily identify and differentiate one from the other.
  */
 @Suppress("UndocumentedPublicClass")
 sealed class Permission {
     abstract val id: String?
     abstract val desc: String?
+    val name: String = this::class.java.simpleName
```

Comment:
I remembered that working on the Fenix migration, we had a bug when `javaClass.simpleName`  [could produce a different string depending on the Kotlin version](https://github.com/mozilla-mobile/android-components/pull/10810/files),  for that reason we used fixed strings, not sure how delicate could be the `name` field for this use case, just a head up :) (no blocking)

Reply:
Very interesting. Thank you for this.
Then I guess to be sure we can use `canonicalName.substringAfterLast('.')`.
Will change to that.